src/test/java/org/opensearch/knn/index/NestedSearchIT.java

```diff
@@ -78,6 +78,30 @@ public void testNestedSearch_whenKIsTwo_thenReturnTwoResults() {
         assertEquals(2, hits.size());
     }
 
+    @SneakyThrows
+    public void testNestedSearchWithFaiss_whenKIsTwo_thenReturnTwoResults() {
+        createKnnIndex(2, KNNEngine.FAISS.getName());
+
+        String doc1 = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+            .add(FIELD_NAME_VECTOR, new Float[] { 1f, 1f }, new Float[] { 1f, 1f })
+            .build();
+        addNestedKnnDoc(INDEX_NAME, "1", doc1);
+
+        String doc2 = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+            .add(FIELD_NAME_VECTOR, new Float[] { 2f, 2f }, new Float[] { 2f, 2f })
+            .build();
+        addNestedKnnDoc(INDEX_NAME, "2", doc2);
+
+        Float[] queryVector = { 1f, 1f };
+        Response response = queryNestedField(INDEX_NAME, 2, queryVector);
+
+        List<Object> hits = (List<Object>) ((Map<String, Object>) createParser(
+            MediaTypeRegistry.getDefaultMediaType().xContent(),
+            EntityUtils.toString(response.getEntity())
+        ).map().get("hits")).get("hits");
+        assertEquals(2, hits.size());
```

Comment:
we are checking here hits.size(), but we should validate the hits.total too. That value in case of nested documents should be equal to the number of parent documents we hit. This is the behavior I observed with Lucene multi vector query.

Reply:
while fixing the nested filters tests, I use the hits.size(), because at that time as we are not handling the multi vector usecase properly while calculating the hits.

Reply:
ack